include/glow/Graph/Log.h

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) 2018-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef GLOW_SUPPORT_LOG_H
+#define GLOW_SUPPORT_LOG_H
+
+#include "llvm/ADT/StringRef.h"
+#include "llvm/ADT/ilist.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/raw_ostream.h"
+
+#include <fstream>
+#include <string>
+
+namespace glow {
+
+class Node;
+
+extern llvm::cl::opt<bool> dumpCompilationLogOpt;
+
+/// A class for logging all compilation related activities.
+class LogContext {
+private:
+  /// A vector that keeps track of all log contents.
+  std::vector<std::string> logContents_;
+
+  /// A vector that keeps track of current compilation scopes.
+  std::vector<std::string> logScopes_;
+
+  /// A string that represents the current full scope name.
+  std::string currentFullScope_ = "";
+
+public:
+  LogContext() = default;
+
+  /// Add content into the contents vector.
+  void addLogContent(llvm::StringRef logContent);
+
+  /// Add a scope into the scope vector.
+  void addLogScope(llvm::StringRef scopeName);
+
+  /// Getter that returns the current full scope name.
+  llvm::StringRef getFullScopeName() { return currentFullScope_; }
+
+  /// Pops out the most recently added scope.
+  void popBackLogScope();
+
+  /// Dumps the log into a file named after the given \p function name.
+  void dumpLog(llvm::StringRef funcName);
+
+  /// Logs the node creation with a list of input nodes.
+  void logNodeCreation(Node *newNode);
```

Comment:
make sure that what can be const is const. Like if you do not intend to modify newNode it needs to be const.
Check all places in this file.